### PR TITLE
Add tests for multi stream usage scenarios

### PR DESCRIPTION
**Pull request by jatinx**
Scenarios Covered:
- Multi Stream Multi Thread
- Multi Stream Multi Thread Multi GPU

### CODE REVIEW

File: tests/catch/stress/stream/streamEnqueue.cc [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+  auto hPtr = std::make_unique<unsigned long long[]>(hwThreads);
+  HIP_CHECK(
+      hipMemcpy(hPtr.get(), dPtr, sizeof(unsigned long long) * hwThreads, hipMemcpyDeviceToHost));
+
+  // Validate that CPU and GPU has the same results
+  for (size_t i = 0; i < hwThreads; i++) {
+    INFO("Check for Index " << i);
+    REQUIRE(hostData[i].data.load() == hPtr[i]);
+  }
+}
```
Comment (chrispaquot) [inline tests/catch/stress/stream/streamEnqueue.cc:114]: you need to free dPtr

File: tests/catch/stress/stream/streamEnqueue.cc [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+      hipStream_t stream{nullptr};
+      HIP_CHECK(hipStreamCreate(&stream));
+      REQUIRE(stream != nullptr);
+      streamPool.push_back(stream);
+
+      int* dPtr{nullptr};
+      HIP_CHECK(hipMalloc(&dPtr, sizeof(int) * size));
+      REQUIRE(dPtr != nullptr);
+      HIP_CHECK(hipMemset(dPtr, 0, sizeof(int) * size));
+      HIP_CHECK(hipDeviceSynchronize());    // Wait till Memset is done
```
Comment (chrispaquot) [inline tests/catch/stress/stream/streamEnqueue.cc]: hipMemset isn't async so you shouldn't need to do a device sync

File: tests/catch/stress/stream/streamEnqueue.cc [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+#endif
+
+  auto enqueueKernelThread = [&]() {
+    for (size_t i = 0; i < maxWorkPerThread; i++) {
+#if HT_NVIDIA
+      std::unique_lock<std::mutex> lock(ness);  // Lock on creation
+#endif
+      hipStream_t stream = streamPool[genStream(engine)];  // Get a random stream
+
+      hipSetDevice(streamToDeviceIndex[stream]);  // Cant do HIP_CHECK on this, catch2 macros and
```
Comment (chrispaquot) [inline tests/catch/stress/stream/streamEnqueue.cc]: you can still check for hipError_t returned value

File: tests/catch/stress/stream/streamEnqueue.cc (hunk 1 above)
Comment (jatinx) [reply tests/catch/stress/stream/streamEnqueue.cc:114]: done

File: tests/catch/stress/stream/streamEnqueue.cc (hunk 2 above)
Comment (jatinx) [reply tests/catch/stress/stream/streamEnqueue.cc]: done

File: tests/catch/stress/stream/streamEnqueue.cc (hunk 3 above)
Comment (jatinx) [reply tests/catch/stress/stream/streamEnqueue.cc]: done

**Review by chrispaquot — APPROVED** [review]

**Pull request merged**